1 - Loading in count data, normalization & QC.Rmd

```diff
@@ -0,0 +1,195 @@
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Wildtype_3), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Wildtype_4), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+```
+
+
+
+Modeling count data for Tumour spheres and Adherent conditions.
+For RNA-seq with 4 treatment and 4 control, treatment = columns 1:4, treatment = control 5:8
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+treatment_mean_counts <- apply(RNA_seq_counts_final[, 1:4], 1, mean) # Control means
+treatment_variance_counts <- apply(RNA_seq_counts_final[, 1:4], 1, var) # Control Variance
+treatment_variance_vs_means <- data.frame(treatment_mean_counts, treatment_variance_counts)
+
+ggplot(treatment_variance_vs_means) +
+  geom_point(aes(x=treatment_mean_counts, y=treatment_variance_counts)) + 
+  geom_line(aes(x=treatment_mean_counts, y=treatment_mean_counts, color="red")) +
+  guides(color=guide_legend(title="Treatment\nvariance = mean")) +
+  scale_y_log10() +
+  scale_x_log10()
+
+control_mean_counts <- apply(RNA_seq_counts_final[, 5:8], 1, mean) # Treatment means
+control_variance_counts <- apply(RNA_seq_counts_final[, 5:8], 1, var) # Treatment variance
+control_variance_vs_means <- data.frame(control_mean_counts, control_variance_counts)
+
+ggplot(control_variance_vs_means) +
+  geom_point(aes(x=control_mean_counts, y=control_variance_counts)) + 
+  geom_line(aes(x=control_mean_counts, y=control_mean_counts, color="red")) +
+  guides(color=guide_legend(title="Control\nvariance = mean")) +
+  scale_y_log10() +
+  scale_x_log10()
+```
+
+
+
+Filter out low read-counts before creating DESeq2Dataset object
+This is done with edgeR's cpm function.
+Rh30 low CPM filtering
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+RNA_seq_edgeR_DEG <- DGEList(RNA_seq_counts_final)
+
+# Calculate normalization factors
+RNA_seq_edgeR_DEG <- calcNormFactors(RNA_seq_edgeR_DEG)
+
+# Filter low-expressed genes, less than 1 cpm (counts per million)
+cpm_cutoff <- 0.2
+
+# Min cpm of each row of cpm(gene_counts), that is less than cutoff
+genes_to_drop <- which(apply(cpm(RNA_seq_edgeR_DEG), 1, min) < cpm_cutoff)
+RNA_seq_cpm_filtered <- RNA_seq_counts_final[-genes_to_drop,]
+```
+
+
+
+Actual DESeq2 part
+
+This is the part which is slightly debatable and probably needs one trial and error to see which works best.
+
+On some biostars forum posts, it is not recommended to do any form of CPM filtering before DESeq2 because it automatically
+does some form internal filtering before fitting its model. For edgeR, this filtering is needed as it is well described in
+their vignette. 
+
+So both options to add CPM filtering or use the raw counts straight into DESeq2 can be tried. For this, the variable "RNA_seq_final" can be swapped for "RNA_seq_counts_filtered" to use the CPM filtered one instead.
```

Comment:
"RNA_seq_counts_final" and "RNA_seq_cpm_filtered" instead!

Reply:
ok